tools/federation/src/main/java/org/eclipse/rdf4j/federated/evaluation/SailTripleSource.java

```diff
@@ -109,9 +112,11 @@ public CloseableIteration<Statement, QueryEvaluationException> getStatements(
 
 		return withConnection((conn, resultHolder) -> {
 
-			RepositoryResult<Statement> repoResult = conn.getStatements(subj, pred, obj,
-					queryInfo.getIncludeInferred(), contexts);
+			RepositoryResult<Statement> repoResult = null;
 			try {
+				repoResult = conn.getStatements(subj, pred, obj,
+						queryInfo.getIncludeInferred(), contexts);
```

Comment:
A few of the fixes are simply to make sure that we always close resources if there is an exception. So mostly moving the variables outside of the try-catch and adding a null check to the catch block.